File: server/src/main/java/com/plantstein/server/model/Species.java
package com.plantstein.server.model;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;


@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Species {

    @Id
    private String name;

    @NotEmpty
    private Double perfectTemperature;

    @NotEmpty
    private Double perfectLight;

    @NotEmpty
    private Double perfectHumidity;

    @NotEmpty
    private String homeland;

   @NotEmpty
   private String botanicalName;

   @NotEmpty
   private String bloomTime;

    @NotNull
    private Double maxHeight;
}

File: lib/species.dart
class Species {
  final String name;
  final String homeland;
  final double perfectTemperature;
  final double perfectMoisture;
  final double perfectLight;
  final double perfectHumidity;
  final String bloomTime;
  final String botanicalName;
  final double maxHeight;

  const Species(
      {
      required this.name,
      required this.homeland,
      required this.perfectTemperature,
      required this.perfectLight,
      required this.perfectHumidity,
      required this.perfectMoisture,
      required this.bloomTime,
      required this.maxHeight,
      required this.botanicalName});

  factory Species.fromJson(Map<String, dynamic> json) => Species(
        name: json['name'],
        homeland: json['homeland'],
        maxHeight: json['maxHeight'],
        perfectTemperature: json['perfectTemperature'],
        perfectHumidity: json['perfectHumidity'],
        perfectLight: json['perfectLight'],
        perfectMoisture: json['perfectLight'],
        bloomTime: json['bloomTime'],
        botanicalName: json['botanicalName']
      );

  Map<String, dynamic> toJson() => {
        'title': name,
        'homeland': homeland,
        'perfectTemperature': perfectTemperature,
        'perfectLight': perfectLight,
        'perfectMoisture': perfectMoisture,
        'perfectHumidity': perfectHumidity,
        'bloomTime': bloomTime,
        'botanicalName': botanicalName,
        'maxHeight': maxHeight,
      };
}
